<?xml version="1.0"?>
<launch>
  <arg name="marker_size" default="4.4" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />
  <arg name="cam_image_topic" default="/ardrone/bottom/image_raw" />
  <arg name="cam_info_topic" default="/ardrone/bottom/camera_info" />
  <arg name="output_frame" default="/ardrone_base_bottomcam" />

  <!-- Connect to drone. -->
  <arg name="droneip" default="192.168.86.198" />
  <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" args="-ip $(arg droneip)">
    <param name="navdata_demo" value="False" />
    <param name="realtime_navdata" value="True" />
    <param name="realtime_video" value="True" />
    <param name="looprate" value="30" />
  </node>

  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size"           type="double" value="$(arg marker_size)" />
    <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
    <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
    <param name="output_frame"          type="string" value="$(arg output_frame)" />

    <remap from="camera_image"  to="$(arg cam_image_topic)" />
    <remap from="camera_info"   to="$(arg cam_info_topic)" />
  </node>

  <!-- Start tag tracking script. -->
  <node name="tag_tracker" pkg="journey" type="tag_tracker.py"
    output="screen" respawn="true"></node>
</launch>
